code/game/objects/items/storage/storage.dm

```diff
@@ -719,4 +746,39 @@
 	var/obj/item/drawn_item = contents[length(contents)]
 	drawn_item.attack_hand(user)
 
+///Called during Initialize this will iterate over spawns_with and spawns_prob to create our initial contents
 /obj/item/storage/proc/PopulateContents()
+	if(LAZYLEN(spawns_with))
+		var/iterations = 0
+		while(iterations < spawns_mult)
+			iterations += 1
+			for(var/item in spawns_with)
+				new item(src)
+
+	if(LAZYLEN(spawns_prob) && spawns_prob_max)
+		var/list/spawnlists = list()
+		var/list/spawnprobs = list()
+		var/index = 1
+		for(var/list/spawnlist in spawns_prob)
+			var/listprob = spawnlist[1]
+			if(prob(listprob))
+				spawnlist -= listprob
+				spawnlists.len += 1
+				spawnlists[index] = spawnlist
+				spawnprobs += listprob
+				index += 1
+
+		var/spawnleft = min(spawns_prob_max, spawnlists.len)
+		while(spawnleft > 0) // This is ugly but needed, byond is fucky
+			spawnleft -= 1
+			var/lowestprob = INFINITY
+			for(var/listprob in spawnprobs)
+				lowestprob = min(lowestprob, listprob)
```

Comment:
just have a var with the lowest index saved here?

Reply:
why is the lower prob prioritized?

Reply:
Lower probability is prioritized so you dont have higher probabilities over shadowing lower probabilities